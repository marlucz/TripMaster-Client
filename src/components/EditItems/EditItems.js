import React from 'react';
import styled, { css } from 'styled-components';

import { breakpoints, theme } from 'theme/GlobalStyle';
import { ReactComponent as Edit } from 'assets/icons/edit.svg';
import { ReactComponent as Delete } from 'assets/icons/delete.svg';

const StyledWrapper = styled.div`
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;

    @media ${breakpoints.md} {
        flex-direction: row;
        justify-content: flex-end;
        align-items: center;
    }
`;

const PinIcon = styled.button`
    background: none;
    border: none;
    cursor: pointer;

    svg {
        width: 2rem;
        height: 2rem;
        margin: 2px;

        path {
            ${({ color }) =>
                color &&
                css`
                    fill: ${theme[color]};
                `}
        }

        @media ${breakpoints.ld} {
            width: 2.5rem;
            height: 2.5rem;
            margin: 0.5rem;
        }
    }
`;

const EditItems = () => (
    <StyledWrapper>
        <PinIcon color="primary">
            <Edit />
        </PinIcon>
        <PinIcon color="secondary">
            <Delete />
        </PinIcon>
    </StyledWrapper>
);

export default EditItems;
