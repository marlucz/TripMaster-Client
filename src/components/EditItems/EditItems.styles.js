import styled, { css } from 'styled-components';

import { breakpoints, theme } from 'theme/GlobalStyle';

export const StyledWrapper = styled.div`
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;

    @media ${breakpoints.md} {
        flex-direction: row;
        justify-content: flex-end;
        align-items: center;
    }
`;

export const PinIcon = styled.button`
    background: none;
    border: none;
    cursor: pointer;

    svg {
        width: 2rem;
        height: 2rem;
        margin: 2px;

        path {
            ${({ color }) =>
                color &&
                css`
                    fill: ${theme[color]};
                `}
        }

        @media ${breakpoints.ld} {
            width: 2.5rem;
            height: 2.5rem;
            margin: 0.5rem;
        }
    }
`;
