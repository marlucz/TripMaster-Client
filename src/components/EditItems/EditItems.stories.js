import React from 'react';
import { storiesOf } from '@storybook/react';

import EditItems from './EditItems';

storiesOf('EditItem', module).add('Normal', () => (
    <EditItems>Hello Marcin</EditItems>
));
