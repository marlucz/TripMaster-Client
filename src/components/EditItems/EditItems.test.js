import React from 'react';
import { render } from '@testing-library/react';
import EditItems from './EditItems';

describe('<EditItems />', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<EditItems />);
        expect(asFragment()).toMatchSnapshot();
    });
});
