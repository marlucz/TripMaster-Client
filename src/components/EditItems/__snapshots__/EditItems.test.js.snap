// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditItems /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm ijPOYZ"
  >
    <button
      class="sc-AxirZ Hklfj"
      color="primary"
    >
      <svg>
        edit.svg
      </svg>
    </button>
    <button
      class="sc-AxirZ bUVRuD"
      color="secondary"
    >
      <svg>
        delete.svg
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
